tests/unit/fem/test_calcvshape.cpp

```diff
@@ -0,0 +1,437 @@
+         T.GetPointMat().SetSize(3, 6);
+         T.GetPointMat()(0, 0) = 0.0;
+         T.GetPointMat()(1, 0) = 0.0;
+         T.GetPointMat()(2, 0) = 0.0;
+         T.GetPointMat()(0, 1) = 1.0;
+         T.GetPointMat()(1, 1) = 0.0;
+         T.GetPointMat()(2, 1) = 0.0;
+         T.GetPointMat()(0, 2) = 0.0;
+         T.GetPointMat()(1, 2) = 1.0;
+         T.GetPointMat()(2, 2) = 0.0;
+         T.GetPointMat()(0, 3) = 0.0;
+         T.GetPointMat()(1, 3) = 0.0;
+         T.GetPointMat()(2, 3) = 1.0;
+         T.GetPointMat()(0, 4) = 1.0;
+         T.GetPointMat()(1, 4) = 0.0;
+         T.GetPointMat()(2, 4) = 1.0;
+         T.GetPointMat()(0, 5) = 0.0;
+         T.GetPointMat()(1, 5) = 1.0;
+         T.GetPointMat()(2, 5) = 1.0;
+         T.SetFE(&WedgeFE);
+         break;
+      default:
+         MFEM_ABORT("Unknown element type \"" << ElemType << "\"");
+         break;
+   }
+}
+
+/**
+ * Tests fe->CalcVShape() over a grid of IntegrationPoints
+ * of resolution res. Also tests at integration points
+ * that are outside the element.
+ */
+void TestCalcVShape(FiniteElement* fe, ElementTransformation * T, int res)
+{
+   int dim = fe->GetDim();
+   int dof = fe->GetDof();
+
+   Vector dofsx(dof);
+   Vector dofsy(dof);
+   Vector dofsz(dof);
+   Vector v(dim);
+   Vector vx(dim); vx = 0.0; vx[0] = 1.0;
+   Vector vy(dim); vy = 0.0;
+   if (dim > 1) { vy[1] = 1.0; }
+   Vector vz(dim); vz = 0.0;
+   if (dim > 2) { vz[2] = 1.0; }
+   DenseMatrix weights( dof, dim );
+
+   VectorConstantCoefficient vxCoef(vx);
+   VectorConstantCoefficient vyCoef(vy);
+   VectorConstantCoefficient vzCoef(vz);
+
+   fe->Project(vxCoef, *T, dofsx);
+   if (dim> 1) { fe->Project(vyCoef, *T, dofsy); }
+   if (dim> 2) { fe->Project(vzCoef, *T, dofsz); }
+
+   // Get a uniform grid or integration points
+   RefinedGeometry* ref = GlobGeometryRefiner.Refine( fe->GetGeomType(), res);
+   const IntegrationRule& intRule = ref->RefPts;
+
+   int npoints = intRule.GetNPoints();
+   for (int i=0; i < npoints; ++i)
+   {
+      // Get the current integration point from intRule
+      IntegrationPoint pt = intRule.IntPoint(i);
+
+      // Get several variants of this integration point
+      // some of which are inside the element and some are outside
+      Array<IntegrationPoint> ipArr;
+      GetRelatedIntegrationPoints( pt, dim, ipArr );
+
+      // For each such integration point check that the weights
+      // from CalcShape() sum to one
+      for (int j=0; j < ipArr.Size(); ++j)
+      {
+         IntegrationPoint& ip = ipArr[j];
+         fe->CalcVShape(ip, weights);
+
+         weights.MultTranspose(dofsx, v);
+         REQUIRE( v[0] == Approx(1.) );
+         if (dim > 1)
+         {
+            weights.MultTranspose(dofsy, v);
+            REQUIRE( v[1] == Approx(1.) );
+         }
+         if (dim > 2)
+         {
+            weights.MultTranspose(dofsz, v);
+            REQUIRE( v[2] == Approx(1.) );
+         }
+      }
+   }
+}
+
+/*
```

Comment:
Is there a reason these tests are commented out?

Reply:
Good question. I think I had plans for these but they are no longer needed. I'll remove them.